posthog/hogql/property.py

```diff
@@ -130,11 +130,19 @@ def property_to_expr(
 
         chain = ["person", "properties"] if property.type == "person" and scope != "person" else ["properties"]
         field = ast.Field(chain=chain + [property.key])
+        properties_field = ast.Field(chain=chain)
 
         if operator == PropertyOperator.is_set:
-            return ast.CompareOperation(op=ast.CompareOperationOp.NotEq, left=field, right=ast.Constant(value=None))
+            return ast.Call(name="JSONHas", args=[properties_field, ast.Constant(value=property.key)])
```

Comment:
Hmm, is this change correct? 

- The `is_not_set` case below is `properties.key == null || key not in properties`
- The `is_set` case now is `key in properties`
- The `is_set` case before was `properties.key != null`

With regards to the `is_not_set` case of `properties.key == null || key not in properties`

I'm kind of surprised that adding `key not in properties` makes a difference, since `properties.key` should return `null` if it's not present.

Then, shouldn't the `is_set` case also be a complete reverse, so `key in properties and properties.key != null` instead of just one? 🤔

Reply:
> Shouldn't the is_set case also be a complete reverse

This would make sense, but that's not how it works in the non-hogql code. This PR makes the hogql code match what we already have. See these:
https://github.com/PostHog/posthog/blob/90d5152a79d3bae2e82191426c01bf82a35dea80/posthog/models/property/util.py#L468
hhttps://github.com/PostHog/posthog/blob/90d5152a79d3bae2e82191426c01bf82a35dea80/posthog/models/property/util.py#L481

Reply:
Cool. I won't block/ask for more then, but... it's possible the previous behaviour was an oversight. If it doesn't look like it'll break any more things, might be worth improving upon the old query? We'll probably find some other weird things as we refactor this codebase 😅